cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -364,6 +364,11 @@ class SimpleArray
 
     ~SimpleArray() = default;
 
+    SimpleArray clone() const
+    {
+        return SimpleArray(*this);
```

Comment:
This is clearly a redundant helper function for C++ consumer, which obviously should use the copy constructor.

The fact that `SimpleArray` is not managed by a `std:shared_ptr` asserts that the copy constructor should be used for copying all contents of the array.

Do not add `SimpleArray::clone()`.  Please remove the helper.  But maybe you can add a line of comment around the copy constructor to explain that we should never add a helper like `clone()`.

Reply:
I see the point. fixed.